File: src/main/org/nlogo/prim/_setbreedvariable.java
package org.nlogo.prim ;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.EngineException;
import org.nlogo.nvm.Syntax;

public final strictfp class _setbreedvariable
	extends Command
{
	private final String name ;
	public _setbreedvariable( _breedvariable original )
	{
		name = original.name ;
	}
	@Override
	public Syntax syntax()
	{
		return Syntax.commandSyntax
			( new int[] { Syntax.TYPE_WILDCARD } ,
			  "-T--" , true ) ;
	}
	@Override
	public String toString()
	{
		return super.toString() + ":" + name ;
	}
	@Override
	public void perform( final Context context )
		throws LogoException
	{
		Object value = args[ 0 ].report( context ) ;
		try
		{
			context.agent.setBreedVariable( name , value ) ;
		}
		catch( org.nlogo.api.AgentException ex )
		{
			throw new EngineException
				( context , this , ex.getMessage() ) ;
		}
		context.ip = next ;
	}
}

File: src/main/prim/_setbreedvariable.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.core.Syntax
import org.nlogo.api.AgentException
import org.nlogo.nvm.{ Command, Context, EngineException }

class _setbreedvariable(name: String) extends Command {

  def this(original: _breedvariable) = this(original.name)

  override def syntax =
    Syntax.commandSyntax(
      right = List(Syntax.WildcardType),
      agentClassString = "-T--",
      switches = true)

  override def toString =
    super.toString + ":" + name

  override def perform(context: Context) {
    val value = args(0).report(context)
    try context.agent.setBreedVariable(name, value)
    catch { case ex: AgentException =>
      throw new EngineException(context, this, ex.getMessage) }
    context.ip = next
  }

}
